src/common/src/error.rs

```diff
@@ -96,6 +96,12 @@ pub enum ErrorCode {
         #[source]
         BoxedError,
     ),
+    #[error("Expr error: {0:?}")]
+    ExprError(
```

Comment:
I'm not sure should we introduce `ExprError(ExprError)` here instead of convert to other RwError.

Reply:
> pub type BoxedError = Box<dyn Error + Send + Sync>;

So there the error is boxed into a trait object. 🤔

`ExprError(ExprError)` looks better to me.

Reply:
I think the expression error will always be capsulated by some other error like `StreamError` when exposed to the user, so there might be no need to make it a variant of `RwError` eventually.
We may keep it for now due to compatibility.

Reply:
> > pub type BoxedError = Box<dyn Error + Send + Sync>;
> 
> So there the error is boxed into a trait object. 🤔
> 
> `ExprError(ExprError)` looks better to me.

It seems we can't refer `ExprError` here due to crate dependencies?

Reply:
> > > pub type BoxedError = Box<dyn Error + Send + Sync>;
> > 
> > 
> > So there the error is boxed into a trait object. 🤔
> > `ExprError(ExprError)` looks better to me.
> 
> It seems we can't refer `ExprError` here due to crate dependencies?

Yes, same as `StorageError`